### PR TITLE
Extend the Digest::UUID module with the clean method

### PR DESCRIPTION
**Pull request by djberg96**
Since Rails 5.1+ gives us the Digest::UUID class, let's extend it with the `clean` method here. The eventual goal is to update the core Hardware class with this method, and remove miq-uuid from gems-pending.

### CODE REVIEW

File: spec/core_ext/digest/uuid_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+describe Digest::UUID do
+  context "clean" do
+    let(:expected) { '01234567-89ab-cdef-abcd-ef0123456789' }
+
+    it "handles a valid lowercase guid as expected" do
```
Comment (Fryguy) [inline spec/core_ext/digest/uuid_spec.rb:5]: Can you remove the `"as expected"`.  Seems redundant to me.

Comment (djberg96) [reply spec/core_ext/digest/uuid_spec.rb]: Updated.

**Comment by djberg96** [comment]
@Fryguy Look ok now?

**Comment by bdunne** [comment]
@Fryguy Any concerns?

**Pull request merged**